src/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_safe_repr.py

```diff
@@ -279,8 +281,54 @@ def _repr_dict(self, obj, level, prefix, suffix,
         yield suffix
 
     def _repr_str(self, obj, level):
-        return self._repr_obj(obj, level,
-                              self.maxstring_inner, self.maxstring_outer)
+        try:
+            if self.raw_value:
+                # For raw value retrieval, ignore all limits.
+                if isinstance(obj, bytes):
+                    yield obj.decode('latin-1')
+                else:
+                    yield self._convert_to_unicode_or_bytes_repr(repr(obj))
+                return
+
+            limit_inner = self.maxother_inner
+            limit_outer = self.maxother_outer
+            limit = limit_inner if level > 0 else limit_outer
+            if len(obj) <= limit:
+                # Note that we check the limit before doing the repr, so, this could account
+                # for up to 3 chars above the limit afterwards. i.e.: u'' or b''
```

Comment:
It can be more than 3 if some characters inside the literal will get escaped. So the most pathological case is a Unicode string where every char is `\U0010ffff`. I don't think we need to worry about that, but it might be worth noting this in the comment so that it doesn't mislead in the future.

Reply:
Done